pkg/controller/job/job_controller.go

```diff
@@ -489,6 +489,13 @@ func (jm *JobController) syncJob(key string) (bool, error) {
 				key, *job.Spec.ActiveDeadlineSeconds)
 			jm.queue.AddAfter(key, time.Duration(*job.Spec.ActiveDeadlineSeconds)*time.Second)
 		}
+
+		// enqueue a sync to check if job past ProgressDeadlineSeconds
```

Comment:
Since this check is only for start move that to the top. I don't want that to be confused with other deadlines.